### PR TITLE
Fix: Query Validation in Files' Admin API

### LINKED ISSUE #18297: Problem with privileges for accessing "Media Library" files
### Required System information

- Node.js version: 18.17.1
- NPM version: 9.6.7
- Strapi version: 4.10.1
- Database: postgres
- Operating system: OS: linux-x64
- Is your project Javascript or Typescript: Javascript

### Describe the bug

I'm having issues accessing files within the "Media Library".

Access privileges to the media library via the "Has the same role as the author" setting do not work

Screenshot 2023-10-04 alle 16 53 48

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

Fixes the query validation for the files' admin API.

It now only validates/sanitizes the request query, not the one generated with permissions/conditions. 

### Why is it needed?

See #18297

### How to test it?

`yarn test:api tests/api/core/upload/admin/file.test.api.js`

or see reproduction steps on #18519

### Related issue(s)/PR(s)

fix #18297

### CODE REVIEW

File: packages/core/upload/server/controllers/admin-file.js [hunk 1]
```diff
@@ -24,10 +24,16 @@ module.exports = {
       return ctx.forbidden();
     }
 
-    const pmQuery = pm.addPermissionsQueryTo(merge(defaultQuery, ctx.query));
+    await pm.validateQuery(ctx.query);
```
Comment (innerdvations) [inline packages/core/upload/server/controllers/admin-file.js:27]:
```suggestion
    // validate the incoming user query params
    await pm.validateQuery(ctx.query);
```

File: packages/core/upload/server/controllers/admin-file.js [hunk 2]
```diff
@@ -24,10 +24,16 @@ module.exports = {
...
 
-    await pm.validateQuery(pmQuery);
-    const query = await pm.sanitizeQuery(pmQuery);
+    const query = await pipeAsync(
+      // Start by sanitizing the incoming query
+      (q) => pm.sanitizeQuery(q),
+      // Add the default query
```
Comment (innerdvations) [inline packages/core/upload/server/controllers/admin-file.js:32]:
```suggestion
      // Add the default query which should not be validated or sanitized
```

File: tests/api/core/upload/admin/file.test.api.js [hunk 3]
```diff
@@ -53,6 +56,81 @@ describe('Upload', () => {
...
+        },
+        {
+          action: 'plugin::upload.assets.update',
+          subject: null,
+          conditions: ['admin::has-same-role-as-creator'],
+          properties: {},
+        },
+      ]);
+
+      // TODO: We create 2 users in order to be able to test the condition itself (same role as creator)
```
Comment (innerdvations) [inline tests/api/core/upload/admin/file.test.api.js:95]:
```suggestion
      // Create 2 users in order to be able to test the condition itself (same role as creator)
```

**Comment by innerdvations** [comment]
I think the failing CI could be because our mysql:latest container has upgraded to 8.4 from 8.3.x

File: tests/api/core/upload/admin/file.test.api.js (hunk 3 above)
Comment (Convly) [reply tests/api/core/upload/admin/file.test.api.js:95]: We're not testing the condition for now, hence the todo, why do you want to remove it?

**Comment by Convly** [comment]
> I think the failing CI could be because our mysql:latest container has upgraded to 8.4 from 8.3.x

It is, I've tried to investigate that in another PR, but no solution yet

**Comment by innerdvations** [comment]
Initial guess is that it's an issue with the `bitnami/mysql:latest` container that they'll fix, but it looks like if we switch to `mysql:latest` [it works](https://github.com/strapi/strapi/pull/20400). But I don't have context for why we're using bitnami.

File: tests/api/core/upload/admin/file.test.api.js (hunk 3 above)
Comment (Convly) [reply tests/api/core/upload/admin/file.test.api.js:95]: The todo is to say that we'll need additional tests to check the condition itself (not just that you can send the query, but also that the condition is respected)

**Review by innerdvations — APPROVED** [review]
LGTM!

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**